src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/CanonicalDefinitionEETypeNode.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    internal sealed class CanonicalDefinitionEETypeNode : EETypeNode
+    {
+        public CanonicalDefinitionEETypeNode(NodeFactory factory, TypeDesc type) : base(factory, type)
+        {
+            Debug.Assert(type.IsCanonicalDefinitionType(CanonicalFormKind.Any));
+        }
+
+        public override bool ShouldSkipEmittingObjectNode(NodeFactory factory) => false;
+        public override bool StaticDependenciesAreComputed => true;
+        public override bool IsShareable => IsTypeNodeShareable(_type);
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory) => null;
+        protected override int GCDescSize => 0;
+
+        protected override void ComputeOptionalEETypeFields(NodeFactory factory)
+        {
+            // TODO: handle the __UniversalCanon case (valuetype padding optional field...)
+            Debug.Assert(_type.IsCanonicalDefinitionType(CanonicalFormKind.Specific));
+        }
+
+        protected override void OutputBaseSize(ref ObjectDataBuilder objData)
+        {
+            // Canonical definition types will have their base size set to the minimum
+            objData.EmitInt(_type.Context.Target.PointerSize * 3);
+        }
+
+        protected override void OutputVirtualSlotAndInterfaceCount(NodeFactory factory, ref ObjectDataBuilder objData)
+        {
+            // TODO: handle the __UniversalCanon case (base type is System.ValueType)
+            // We emit a vtalbe with ToString, GetHashCode and Equals for valuetypes in ProjectN)
+            Debug.Assert(_type.IsCanonicalDefinitionType(CanonicalFormKind.Specific));
+
+            var baseType = _type.BaseType;
+            Debug.Assert(baseType != null && factory.VTable(baseType).Slots.Count == 3);
```

Comment:
I believe the type loader needs to be able to resolve these EEType handles, that's why we need to allocate them.
My goal was to emit the same data we emitted on ProjectN for these types, but you are right that we'll never use the vtable for these 2 special types. I'll remove that.